src/server/modules/job/services/JobManagementService.ts

```diff
@@ -104,6 +109,64 @@ class JobManagementService implements interfaces.JobManagementService {
     const currJobStatus = this.computeJobStatus(jobItems)
     return this.jobRepository.update(job, { status: currJobStatus })
   }
+
+  getJobInformation: (jobId: number) => Promise<interfaces.JobInformation> =
```

Comment:
haha yes I thought the same and I looked briefly into it but there's Typescript issues with creating private methods on interfaces

https://github.com/microsoft/TypeScript/issues/25163

I didn't want to attempt too much refactoring, but if you've seen an elegant solution, please share it!

Reply:
could we afford to remove `computeJobStatus` in the interface, which should allow the class's `computeJobStatus` to remain as a private method?

Reply:
was thinking that `getJobInformation` could just be removed from the interface entirely, but yea looks like that would cause TS issues with the unit tests :( am okay with this then!